### ISSUE TITLE
Replaced usage of dicts with nonsense values with sets in select reactor

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6784]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Alex reported|
|-|-|
|Trac ID|trac#6784|
|Type|enhancement|
|Created|2013-10-10 15:13:00Z|
|Branch|https://github.com/twisted/twisted/tree/select-sets-6784|

This should save memory, and with it a bit of CPU time.

Attachments:

* [t6784.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/01e/01e7449647c0a73179a4076b1f5088ff677b188f/0c90a8454f1242dd2cfebec42aef74f3e6cd4c24.diff) (3920 bytes) - added by Alex on 2013-10-10 15:13:16Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6784 6784
type__enhancement enhancement
reporter__Alex Alex
priority__normal normal
milestone__ 
branch__branches_select_sets_6784 branches/select-sets-6784
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1381417980000000 1381417980000000
changetime__1387025939000000 1387025939000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [40958]) Branching to 'select-sets-6784'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [40960]) apply t6784.diff

refs [#6784](https://github.com/twisted/twisted/issues/6784)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @exarkun|
|-|-|

Thanks.  Looks good to me and buildbot seems to agree.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [40979]) Merge select-sets-6784

Author: Alex
Reviewer: exarkun
Fixes: [#6784](https://github.com/twisted/twisted/issues/6784)

Use the set type instead of the dict type for the readers and writers in selectreactor
since they are logically more like sets than mappings.
